util/ranger/ranger.go

```diff
@@ -270,6 +270,10 @@ func buildColumnRange(accessConditions []expression.Expression, sc *stmtctx.Stat
 				ran.HighExclude = false
 			}
 		}
+		ranges, err = unionRanges(sc, ranges)
```

Comment:
Only if `colLen != types.UnspecifiedLength` should union the ranges?

Reply:
Yes, only cuting by prefix will cause the case the ranges are overflapped.
`a in ('aaa', 'aab', 'aac')`
If the len is 1. The ranges after cutting before unioning is `[a, a], [a, a], [a, a]`.